Libraries/IonAuth.php

```diff
@@ -292,6 +292,86 @@ public function register(string $identity, string $password, string $email, arra
 		}
 	}
 
+	/**
+	 * Send activation email.
+	 *
+	 * @param string $identity
+	 *
+	 * @return boolean|array return an array of activation details if CI e-mail validation is enabled
+	 * @author Ali Ragab
+	 */
+	public function SendActivationEmail(string $identity): bool|array
+	{
+		if (empty($identity)) {
+			$this->setError('IonAuth.empty_identity');
+			return FALSE;
+		}
+
+		if (!$this->ionAuthModel->identityCheck($identity)) {
+			$this->setError("IonAuth.unregistered_identity");
+			return FALSE;
+		}
+
+		// Retrieve user information
+		$user = $this->where($this->ionAuthModel->identityColumn, $identity)
+			->limit(1)
+			->users()->row();
+
+		if ($user->active) {
+			$this->setError("IonAuth.already_activated_identity");
+			return FALSE;
+		}
+
+		if ($user) {
+
+			// deactivate so the user must follow the activation flow
+			$deactivate = $this->deactivate($user->id);
```

Comment:
Is this (and the next few lines) needed? You're throwing an error above if the user is active.

Reply:
You are right.
Not logic to keep `if ($user) {` as we checked the identity against database and the user must be founded.

But my point regarding `$deactivate = $this->deactivate($user->id);` is to change the activation code every time we request activation code sending. Am I right?

Reply:
Good point.  You could call the model's deactivate method directly to separate those concerns.